### PR TITLE
Add LSP for Terraform

### PR DESCRIPTION
**Pull request by howdoicomputer**
Emacs LSP has support for Terraform these days.

### CODE REVIEW

File: modules/tools/terraform/config.el [hunk 1]
```diff
@@ -13,3 +13,8 @@
   :after terraform-mode
   :config
   (set-company-backend! 'terraform-mode 'company-terraform))
+
+
+(after! terraform-mode
+    (if (featurep! +lsp)
+      (add-hook 'terraform-mode-local-vars-hook #'lsp!))
```
Comment (hlissner) [inline modules/tools/terraform/config.el]: 1. No need for an after! block for a hook.
2. Move it to the top, above the `map!` call, to keep like config together.
3. Use `when` when there is no else clause.
4. The indentation is off.

e.g.

```elisp
(when (featurep! +lsp)
  (add-hook 'terraform-mode-local-vars-hook #'lsp!))
```

Comment (howdoicomputer) [reply modules/tools/terraform/config.el]: Alright! Learning elisp.

**Comment by hlissner** [comment]
Looks good to me. Thanks for the help and quick response!

File: modules/tools/terraform/config.el [hunk 2]
```diff
@@ -1,5 +1,9 @@
 ;;; tools/terraform/config.el -*- lexical-binding: t; -*-
 
+(when (featurep! +lsp)
+  (add-hook 'terraform-mode-local-vars-hook #'lsp!)
```
Comment (robsonpeixoto) [inline modules/tools/terraform/config.el]: Missing a `)`

Comment (howdoicomputer) [reply modules/tools/terraform/config.el]: https://github.com/hlissner/doom-emacs/pull/2566

**Pull request merged**